test/cpp/jit/test_gpu.cpp

```diff
@@ -15680,6 +15889,87 @@ TEST(NVFuserTest, FusionIssue1021_CUDA) {
   testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, TestSimpleWarpPadFail_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto tv0 = makeSymbolicTensor(2);
+
+  fusion->addInput(tv0);
+
+  auto tv1 = sum(tv0, {1});
+  auto tv2 = broadcast(tv1, {false, true});
+  auto tv3 = add(tv2, tv0);
+
+  fusion->addOutput(tv3);
+
+  // Schedule a persistent kernel
+  // auto tv0_cache = tv0->cache_after();
+  // tv1->split(1, 8, false);
+  // auto tv1_rf = tv1->rFactor({1});
+  // tv1_rf->axis(0)->parallelize(ParallelType::BIDx);
+  // tv1_rf->axis(-1)->parallelize(ParallelType::TIDx);
+  tv1->axis(-1)->parallelize(ParallelType::TIDx);
+  tv1->axis(-1)->padToMultipleOfWarp();
+  // TransformPropagator::from(tv1_rf);
+  // tv0->axis(-1)->parallelize(ParallelType::TIDx);
+  // tv0_cache->axis(-1)->parallelize(ParallelType::TIDx);
+  tv2->axis(-1)->parallelize(ParallelType::TIDx);
+  tv3->axis(-1)->parallelize(ParallelType::TIDx);
+
+  // tv0->computeAt(tv3, -1, ComputeAtMode::MostInlined);
+
+  tv1->axis(0)->parallelize(ParallelType::TIDy);
+  tv2->axis(0)->parallelize(ParallelType::TIDy);
+  tv3->axis(0)->parallelize(ParallelType::TIDy);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input1 = at::randn({16, 31}, options);
+
+  auto at_output = input1.sum({1}, true).add(input1);
+
+  FusionExecutor fe;
+  fe.compileFusion(fusion.get());
+  auto outputs = fe.runFusion({input1});
+  testValidate(
+      fusion.get(), outputs, {input1}, {at_output}, __LINE__, __FILE__);
+}
+
+TEST(NVFuserTest, TestSimpleWarpPadFail2_CUDA) {
```

Comment:
Probably needs to be renamed as it doesn't fail anymore.

Reply:
Updated, thanks!